test/built-ins/Iterator/prototype/take/argument-validation-failure-closes-underlying.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024 Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-iteratorprototype.take
+description: >
+  Underlying iterator is closed when argument validation fails
+info: |
+  %Iterator.prototype%.take ( limit )
+
+features: [iterator-helpers]
+flags: []
+---*/
+
+let closed = false;
+let closable = {
+  __proto__: Iterator.prototype,
+  get next() {
+    throw new Test262Error('next should not be read');
+  },
+  return() {
+    closed = true;
+  },
+};
+
+assert.throws(RangeError, function() {
+  closable.take();
+});
```

Comment:
Since the return value of `return` is not an object, wouldn't `TypeError` be thrown before `RangeError` due to step 7 of `IteratorClose`?

Reply:
Both `Iterator#map` and `Iterator#filter`, etc pass the tests, but they probably do not behave as intended. In all tests, the `return` method of `closable` should return an object, I think.